test/signup.test.js

```diff
@@ -0,0 +1,69 @@
+const chai = require('chai');
+const app = require('../index');
+const request = require('supertest')(app);
+const User = require('../models/User');
+const mocha = require('mocha');
+
+const describe = mocha.describe;
+const beforeAll = mocha.before
+
+const expect = chai.expect; 
+describe('POST requests', function() {
+    beforeEach(function(done){
+        User.remove({}, function(err){
+          done();
+        });
+        
+    });
+     describe('/api/users/', function(){
+        
+        describe('status code', function(){
+            it('should be equal to 201', function(done){
+                request.post('/api/users/')
+                .send({
+                    user: {
+                        email: 'goodemailformat@goodemail.com',
+                        username: 'username101',
+                        password: 'passthat1password',
+                    }
+                })
+                .expect(201, done);
+            });
+        })
+        describe('request body', function(){
+            it('should have a data property that contains a token string property', function(done){
+                request.post('/api/users/')
+                .send({
+                    user: {
+                        email: 'goodemailformat@goodemail.com',
+                        username: 'username101',
+                        password: 'passthat1password',
+                    }
+                })
+                .end(function(err, resp){
+                    expect(resp.body).property('data')
+                        .to.have.property('token')
+                    done();
+                });
+            });
+
+            it('should have a data property that contains a username and email  properties', function(done){
+                request.post('/api/users/')
+                .send({
+                    user: {
+                        email: 'aaazbsa@chdcdscci.com',
+                        username: 'abaasz',
+                        password: 'passspcddsdcacsspass',
+                    }
+                })
+                .end(function(err, resp){
+                    expect(resp.body).property('data')
+                        .to.have.property('email');
+                        expect(resp.body).property('data')
+                        .to.have.property('username');
+                    done();
+                });
+            });
+        })
+     });
```

Comment:
don't you think it'd be nice to test when a user tries to signup without inputing any details or some invalid details or a user tries to signup but they were already members. What do you think ?

Reply:
I think that would be nice

Reply:
I would add that